Revit_Core_Engine/Query/DoesIntersect.cs

```diff
@@ -0,0 +1,156 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using Autodesk.Revit.DB;
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Check if two elements intersect.")]
+        [Input("element1", "First element to check the intersection for.")]
+        [Input("element2", "Second element to check the intersection for.")]
+        [Output("bool", "Result of the intersect checking.")]
+        public static bool DoesIntersect(this Element element1, Element element2)
+        {
+            if (element1 == null || element2 == null)
+            {
+                return false;
+            }
+
+            Transform transform1 = element1.Document.LinkTransform() ?? Transform.Identity;
+            Transform transform2 = element2.Document.LinkTransform() ?? Transform.Identity;
+
+            //null or equal transforms
+            if ((transform1.IsIdentity && transform2.IsIdentity) ||  transform1.AlmostEqual(transform2))
+            {
+                ElementIntersectsElementFilter intersectFilter = new ElementIntersectsElementFilter(element2);
+                return intersectFilter.PassesFilter(element1);
+            }
+            else
+            {
+                //host vs link
+                if (transform1.IsIdentity && !transform2.IsIdentity)
+                {
+                    return DoesIntersectWithTransform(element1, element2, transform2);
+                }
+                //link vs host
+                else if (!transform1.IsIdentity && transform2.IsIdentity)
+                {
+                    return DoesIntersectWithTransform(element2, element1, transform1);
+                }
+                //link vs link
+                if (!transform1.IsIdentity && !transform2.IsIdentity)
+                {
+                    Transform doubleTransform = transform2.Multiply(transform1.Inverse);
+                    return DoesIntersectWithTransform(element1, element2, doubleTransform);
+                }
```

Comment:
My gut feeling tells me this could be written in 2 lines:
```suggestion
                Transform doubleTransform = transform2.Multiply(transform1.Inverse);
                return DoesIntersectWithTransform(element1, element2, doubleTransform);
```

Reply:
Wow, I see that 😃